stix2/exceptions.py

```diff
@@ -49,3 +49,10 @@ class ImmutableError(STIXError, ValueError):
 
     def __init__(self):
         super(ImmutableError, self).__init__("Cannot modify properties after creation.")
+
+
+class VersioningError(STIXError, ValueError):
+    """Execption while using the Versioning API"""
+
+    def __init__(self, msg):
```

Comment:
I want to have exceptions that have meaningful attributes (other than just the string message) where they make sense. For example, the list of properties that the user tried to modify in the new version that aren't allowed. 

I don't think we need a VersioningError layer in a hierarchy, but could be convinced otherwise. 

Yes, I mean "property" every time I say "field".